### ISSUE TITLE
merge.conflictStyle zdiff3 yields error

### ISSUE THREAD

**Issue by taladar** [libgit2/libgit2#6189]
### Reproduction steps

Use git 2.35 and set the new conflictstyle zdiff3

```
git config --global merge.conflictStyle zdiff3
```

### Expected behavior

Operations still work even if the zdiff3 conflict style might be substituted for some default value. Ideally it would be supported of course.

### Actual behavior

This happens when calling e.g. cargo-deny which uses git2-rs which uses libgit2

```
2022-01-26 10:53:03 [ERROR] failed to fetch advisory database: git operation failed: unknown style 'zdiff3' given for 'merge.conflictstyle'; class=Checkout (20)
```

### Version of libgit2 (release number or SHA1)

not quite sure but I assume the most recent release

I first reported this here on git2-rs

https://github.com/rust-lang/git2-rs/issues/795

### Operating system(s) tested

Gentoo Linux

**Comment by ethomson** [comment]
This is certainly a worthwhile feature - but also, it feels like cargo-deny might want to take more control over the way in which it gets/writes/etc advisory databases.  It feels like this shouldn't be impacted by a user's git config settings.

**Comment by taladar** [comment]
I am not the author, just a user.

> It feels like this shouldn't be impacted by a user's git config settings.

I agree. I assume it is read as part of a pull operation but I haven't looked into it in detail. For something like that (pure consumption of upstream git repository of data) what would probably be more appropriate would be a fetch + reset --hard

**Comment by ethomson** [comment]
Fixed via #6195
